paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/source/operator/TestingSourceOperator.java

```diff
@@ -0,0 +1,175 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink.source.operator;
+
+import org.apache.paimon.flink.source.SimpleSourceSplit;
+import org.apache.paimon.flink.source.SimpleSourceSplitSerializer;
+
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.common.state.OperatorStateStore;
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.fs.CloseableRegistry;
+import org.apache.flink.runtime.execution.Environment;
+import org.apache.flink.runtime.metrics.groups.UnregisteredMetricGroups;
+import org.apache.flink.runtime.operators.coordination.OperatorEvent;
+import org.apache.flink.runtime.operators.coordination.OperatorEventGateway;
+import org.apache.flink.runtime.operators.testutils.DummyEnvironment;
+import org.apache.flink.runtime.operators.testutils.MockEnvironmentBuilder;
+import org.apache.flink.runtime.state.AbstractStateBackend;
+import org.apache.flink.runtime.state.OperatorStateBackendParametersImpl;
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.runtime.state.StateInitializationContextImpl;
+import org.apache.flink.runtime.state.hashmap.HashMapStateBackend;
+import org.apache.flink.streaming.api.operators.SourceOperator;
+import org.apache.flink.streaming.api.operators.StreamOperatorParameters;
+import org.apache.flink.streaming.api.operators.StreamingRuntimeContext;
+import org.apache.flink.streaming.runtime.tasks.ProcessingTimeService;
+import org.apache.flink.streaming.runtime.tasks.SourceOperatorStreamTask;
+import org.apache.flink.streaming.runtime.tasks.TestProcessingTimeService;
+import org.apache.flink.streaming.util.MockOutput;
+import org.apache.flink.streaming.util.MockStreamConfig;
+import org.apache.flink.streaming.util.MockStreamingRuntimeContext;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+/** A SourceOperator extension to simplify test setup. */
+public class TestingSourceOperator<T> extends SourceOperator<T, SimpleSourceSplit> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final int subtaskIndex;
+    private final int parallelism;
+
+    public TestingSourceOperator(
+            StreamOperatorParameters<T> parameters,
+            SourceReader<T, SimpleSourceSplit> reader,
+            WatermarkStrategy<T> watermarkStrategy,
+            ProcessingTimeService timeService,
+            boolean emitProgressiveWatermarks) {
+
+        this(
+                parameters,
+                reader,
+                watermarkStrategy,
+                timeService,
+                new TestingOperatorEventGateway(),
+                1,
+                5,
```

Comment:
What are these magic numbers?

Reply:
`TestingSourceOperator` is introduced in reference to the Flink class with the same name. These numbers should be randomized values used to initialize MockRuntimeContext. I have added reference to that Flink class in the Javadoc.